metadata/metadata.go

```diff
@@ -119,6 +119,46 @@ func CopyLabelsWithPrefixReplacement(source, destination v1.Object, sourcePrefix
 	return nil
 }
 
+// DeleteAnnotation deletes the annotation specified by name from the referenced object
+// if the annotation doesn't exist it's a no-op
+func DeleteAnnotation(obj v1.Object, key string) error {
+	if obj == nil {
+		return errors.New("object cannot be nil")
+	}
+
+	annotations := obj.GetAnnotations()
+	if annotations == nil {
+		return nil // nothing to delete
+	}
+
+	if _, ok := annotations[key]; ok {
```

Comment:
I don't think you need this check. Delete takes care of it already, isn't it?

Reply:
true